contracts/utils/hardhat-helpers.js

```diff
@@ -114,12 +114,14 @@ const getHardhatNetworkProperties = () => {
   }
 
   let provider = providerUrl;
-  if (isArbForkTest) {
-    provider = arbitrumProviderUrl;
-  } else if (isHoleskyForkTest) {
-    provider = holeskyProviderUrl;
-  } else if (isBaseForkTest) {
-    provider = baseProviderUrl;
+  if (!providerUrl.includes("localhost")) {
+    if (isArbForkTest) {
+      provider = arbitrumProviderUrl;
+    } else if (isHoleskyForkTest) {
+      provider = holeskyProviderUrl;
+    } else if (isBaseForkTest) {
+      provider = baseProviderUrl;
+    }
```

Comment:
@shahthepro perhaps it would be cool to add smth like this here: 
```
if (isFork){
if (isArbForkTest) {
      provider = arbitrumProviderUrl;
    } else if (isHoleskyForkTest) {
      provider = holeskyProviderUrl;
    } else if (isBaseForkTest) {
      provider = baseProviderUrl;
    } else {
       throw new Error("Non configured network")
    }
}
```